File: benchmark/workloads/javascript/objectArray.js
class Box {
  constructor(value) {
    this.value = value;
  }
}

const values = new Array(200_000);
for (let index = 0; index < values.length; index += 1) values[index] = new Box(index + 1);
let total = 0;
for (let index = 0; index < values.length; index += 1) total += values[index].value;
console.log(total);

File: benchmark/workloads/rust/objectArray.rs
struct Item {
    value: u64,
}

fn main() {
    let mut values = Vec::with_capacity(200_000);
    for index in 1..=200_000_u64 {
        values.push(Box::new(Item { value: index }));
    }
    let total: u64 = values.iter().map(|item| item.value).sum();
    println!("{total}");
}
